web/models.py

```
@@ -2804,3 +2804,193 @@ class ScheduledPost(models.Model):
+        return self.events.filter(start_time__gte=timezone.now()).order_by('start_time')
+
+
+class ChapterMembership(models.Model):
+    """Membership and roles within a chapter."""
+    
+    ROLE_CHOICES = [
+        ('lead', 'Chapter Lead'),
+        ('co_organizer', 'Co-Organizer'),
+        ('volunteer', 'Volunteer'),
+        ('participant', 'Participant'),
+    ]
```

Ruff (0.8.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)